### ISSUE TITLE
Sync conflicts for a single user, maybe due folder renaming

### ISSUE THREAD

**Issue by mibadim** [owncloud/client#5037]
### Expected behaviour

When being the only user signed up on the server, there should happen any sync conflicts and therefore no `*_conflict` files being produced at all.
### Actual behaviour

Sometimes, I suddenly get lots of sync conflicts. This way ownCloud periodically messes up my projects. This is annoying.
### Steps to reproduce

I was not able to reproduce it. However, the last time it appeared I renamed a Java package `some_package` to `somePackage` and let my IDE update all references. One day later, I suddenly got lots of `_conflict` files. In the (new) folder `somePackage`:

```
$ ls
DegreeHeap.java
DegreeHeapTest.java
GraphFinderThread.java
GraphGUI.java
InfoPanel.java
ParamsFromGUI.java
Ranking.java
SideBar_conflict-20160705-123900.java
SideBar.java
SnakeGraph.java
style.css
style.css~
ThreadManager_conflict-20160702-113919.java
ThreadManager.java
VisualGraph.java
WorstGraphFinder_conflict-20160705-120808.java
WorstGraphFinder.java
```

And -- most interesting -- the old version of a conflict file is now in the renamed folder `somePackage`, but still including with a reference to the old package `some_package`:

```
$ head ThreadManager.java 
package some_package; 

import ...
```

while the `_conflict` file seems to be the -- correct -- latest version:

```
$ head ThreadManager_conflict-20160702-113919.java 
package somePackage;
```

Here, `somePackage` is the only folder affected by that behaviour.
### Server configuration

Operating system: Debian 8

Web server: Apache 2.4.10

Database: 5.5.46-0+deb7u1

PHP version: 5.6.20-0+deb8u1

ownCloud version: 8.1.8-12.1

Storage backend:
### Client configuration

Client version: 2.2.2-18.1

Operating system: Debian 8

OS language: German

Installation path of client:

`/usr/bin/`
### Logs

Please use Gist (https://gist.github.com/) or a similar code paster for longer
logs.

`Template for output < 10 lines`
1. Output of `owncloud --logwindow` or `owncloud --logfile log.txt`

see [here](https://gist.github.com/mibadim/4e877fc5ce62b23933001e0a909432e1#file-gistfile1-txt)
2. Web server error log: 

(nothing suspicious)
1. ownCloud log (data/owncloud.log):

~~(could not find this file, sorry!)~~ [Here](https://gist.github.com/mibadim/197fe661aac7c4222c98b0408f2e4933#file-owncloud-log) it is.

**Comment by guruz** [comment]
The owncloud.log is on the server inside the ```data/``` directory. You can also see it from the admin interface.

Not sure how those conflicts can happen these days.

FYI @PVince81

**Comment by PVince81** [comment]
A single user can still have multiple clients.

I have a single account on my ownCloud but sometimes I edit files offline on two different devices (not realizing I was offline) and later on it produces expected conflicts.

**Comment by mibadim** [comment]
I use ownCloud as the only user on exactly one single device, however, I still get conflicts.
By the way: Some files were irrecoverably overwritten by an old version **without** any conflict file. This makes using ownCloud even worse than not doing any backup at all.

I added my `owncloud.log`. It is ~13 MB large, have fun ;-).

**Comment by PVince81** [comment]
@mibadim thanks for the clarification.

Did you enable file locking on 8.1.8 ? See https://doc.owncloud.org/server/8.1/admin_manual/configuration_files/files_locking_transactional.html (requires redis)

Or update to 8.2.6 which has it enabled by default and doesn't require redis.

Since you only have a single client then the only concurrency I can think of is parallel uploads.

**Comment by oneup40** [comment]
I just saw this issue as well. Similar to @mibadim I am the only person using the ownCloud server and use it primarily to keep things synced between my laptop and desktop. My laptop has been powered off all day and was synced last night before I shut it down, so it shouldn't have been able to do anything that would have caused a conflict.

I've been working on a C++ project on my desktop for the past hour or two with no problems. Over that time there is one header file in particular I've edited and saved several times. The most recent time I saved it, the ownCloud client popped up and said there was a sync conflict on that file, moved my most recent file to who knows where, and wrote over my changes with the previously saved version.

At first the sync conflict showed up under the "Not Synced" tab, but in the 5 minutes I've taken to write  this comment, it's disappeared from there. The old version is still on disk though and now I have no clue where the client moved the newest version since there is no indication of a problem under the "Server Activity", "Sync Protocol", or "Not Synced" tabs.

This was fairly light usage, the server isn't under any load, and everything is on a local, quiet LAN.
